src/modules/claim/claim.service.ts

```diff
@@ -379,4 +371,23 @@ export class ClaimService {
     const { sub: subject } = jwt.decode(token);
     return v5(JSON.stringify({ subject, role, version }), UUID_NAMESPACE);
   }
+
+  private async filterUserRelatedClaims(currentUser: string, qb: SelectQueryBuilder<Claim>) {
+    const ownedAssets = (await this.assetService.getByOwner(currentUser)).map((a) => a.id);
+    const offeredAssets = (await this.assetService.getByOfferedTo(currentUser)).map((a) => a.id);
+
+    qb.andWhere(new Brackets(query => {
+      query.where(':currentUser = ANY (claim.claimIssuer)', {
+        currentUser,
+      })
+        .orWhere('claim.subject = :currentUser', { currentUser })
+        .orWhere('claim.requester = :currentUser ', { currentUser });
+      if (ownedAssets.length > 0) {
+        query.orWhere('claim.subject IN (:...ownedAssets)', { ownedAssets });
+      }
+      if (offeredAssets.length > 0) {
+        query.orWhere('claim.subject IN (:...offeredAssets)', { offeredAssets });
+      }
+    }));
```

Comment:
I think this might not work for this new approach. I mean, if we are not adding the `claim.claimIssuer` ahead of time, wouldn't none of these case be met when we are trying to dynamically query the issuer?

Reply:
As I understand, only first clause of query verifies claim issuer, while others doesn't depend on result of this verification